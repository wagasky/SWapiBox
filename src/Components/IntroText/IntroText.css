.main {
  width: 100%;
  height: 100%;
  background: #000;
  overflow: hidden;
}

/*  .start {
    font-size: 200%;
    $start-width: 14em;
    width: $start-width;
    margin: -4em 0 0 (- $start-width / 2);
    text-align: center;
    cursor: pointer;
    
    span {
      color: rgb(75, 213, 238); 
    }
  }    
  
  .intro {
    $intro-width: 15em;
    width: $intro-width;
    margin: 0 0 0 (- $intro-width / 2);
    font-size: 200%;
    font-weight: 400;
    color: rgb(75, 213, 238);
    opacity: 0;
    animation: intro 6s ease-out 1s;
  }
  
  .logo {
    opacity: 0;
    animation: logo 9s ease-out 9s; 
    
    svg {
      width: inherit;
    }
  }*/
  
.main h1 {
  font-size: 2em;
}

.fade {
  position: relative;
  width: 100%;
  min-height: 300px;
  top: -25px;
  background-image: linear-gradient(0deg, transparent, black 75%);
  z-index: 1;
}

.star-wars {
  display: flex;
  justify-content: center;
  position: relative;
  height: 200px;
  color: #feda4a;
  font-family: 'Pathway Gothic One', sans-serif;
  font-size: 350%;
  font-weight: 600;
  /*letter-spacing: 0px;*/
  line-height: 150%;
  perspective: 250px;
  text-align: justify;
}

.crawl {
  position: relative;
  top: 9999px;
  transform-origin: 50% 100%;
  animation: crawl 80s linear;
}

.crawl > .title {
  font-size: 90%;
  text-align: center;
}

.crawl > .title h1 {
  margin: 0 0 100px;
  text-transform: uppercase;
}

.crawl p {

}

@keyframes crawl {
  0% {
    top: 0;
    transform: rotateX(25deg)  translateZ(0);
  }
  100% { 
    top: -6000px;
    transform: rotateX(25deg) translateZ(-2500px);
  }
}