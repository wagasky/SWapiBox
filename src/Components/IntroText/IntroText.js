import React from 'react';
import propTypes from 'prop-types';
import './IntroText.css'

const IntroText = ({ film }) => {

  const { title, releaseYear, numeral, text } = film;

  return (
    <div className="main">

      <div className="fade"></div>
        <audio preload="auto">
          <source src="https://s.cdpn.io/1202/Star_Wars_original_opening_crawl_1977.ogg" type="audio/ogg" />
          <source src="https://s.cdpn.io/1202/Star_Wars_original_opening_crawl_1977.mp3" type="audio/mpeg" />
        </audio>
  
      <section className="star-wars">
      <div className="crawl">
       <div className="title">
          <h4>Episode {numeral}</h4>
          <h1>{title}</h1>
          <h4>{releaseYear}</h4>
        </div>
          <p>{ text }</p> 
        </div>
     </section>
    </div>
  );
};

IntroText.propTypes = {
  film: propTypes.oneOfType([propTypes.array, propTypes.object]),
};

export default IntroText;

// <audio src{Theme} autoPlay />