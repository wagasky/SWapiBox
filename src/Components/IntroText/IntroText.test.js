/* eslint-disable */ 

import React from 'react';
import ReactDOM from 'react-dom';
import IntroText from './IntroText';
import { shallow } from 'enzyme';

it.skip('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<IntroText />, div);
  ReactDOM.unmountComponentAtNode(div);
});