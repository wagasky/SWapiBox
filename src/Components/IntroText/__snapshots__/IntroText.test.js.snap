// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IntroText should match snapshot 1`] = `
<div
  className="main"
>
  <div
    className="fade"
  />
  <audio
    preload="auto"
  >
    <source
      src="https://s.cdpn.io/1202/Star_Wars_original_opening_crawl_1977.mp3"
      type="audio/mpeg"
    />
  </audio>
  <section
    className="star-wars"
  >
    <div
      className="crawl"
    >
      <div
        className="title"
      >
        <h4>
          Episode 
        </h4>
        <h1 />
        <h4 />
      </div>
      <p />
    </div>
  </section>
</div>
`;
